src/host-session-service.vala

```diff
@@ -345,6 +345,29 @@ namespace Frida {
 			throw new Error.INVALID_OPERATION ("Only meant to be implemented by services");
 		}
 
+		public async HashTable<string, Variant>? query_system_parameters (Cancellable? cancellable) throws Error, IOError {
+			#if DARWIN && !IOS
+				Variant? platform = new Variant ("s", "macos");
+			#elif IOS
+				Variant? platform = new Variant ("s", "ios");
+			#elif ANDROID && (X86 || X86_64)
+				Variant? platform = new Variant ("s", "android_emulated");
+			#elif ANDROID
+				Variant? platform = new Variant ("s", "android");
+			#elif LINUX
+				Variant? platform = new Variant ("s", "linux");
+			#elif WINDOWS
+				Variant? platform = new Variant ("s", "windows");
+			#else
+				Variant? platform = new Variant ("s", "unknown");
+			#endif
+			
+			var system_parameters = new HashTable<string, Variant> (str_hash, str_equal);
+			system_parameters.insert ("platform", platform);
```

Comment:
```suggestion
			system_parameters["os"] = os;
```

To make it more specific than `Process.platform` and avoid confusion that it's named the same but has different granularity.